### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by marcelotrevisani**

### CODE REVIEW

**Review by nicoddemus — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -48,12 +48,12 @@ jobs:
       with:
         python-version: ${{ matrix.python }}
 
-    - name: Install tox
+    - name: Install pkgs
       run: |
         python -m pip install --upgrade pip
-        pip install tox
+        python -m pip install -e .[testing]
```
Comment (nicoddemus) [inline .github/workflows/test.yml:54]: In line with https://github.com/gabrielcnr/pytest-datadir/pull/70/files#r1365906429, we really just need `tox` here: `tox` will then be responsible for installing the pre-existing package and any required dependencies.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -48,12 +48,12 @@ jobs:
...
 
     - name: Test
       shell: bash
       run: |
-        tox run -e py --installpkg `find dist/*.tar.gz`
```
[the remark this answers is not in the document]
Comment (marcelotrevisani) [reply .github/workflows/test.yml:59]: sorry for my delay, I forgot about this PR, I created it when I was on the train 😄 
I was just doing one test, gonna revert it back, thanks  :)

File: .github/workflows/test.yml (hunk 1 above)
Comment (marcelotrevisani) [reply .github/workflows/test.yml]: done

File: setup.py [hunk 3]
```diff
@@ -1,42 +1,4 @@
...
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-        "Programming Language :: Python :: 3.11",
-        "Programming Language :: Python :: 3.12",
-        "Topic :: Software Development :: Quality Assurance",
-        "Topic :: Software Development :: Testing",
-    ],
-)
+# Please check pyproject.toml for metadata
+setup(name="pytest-datadir")
```
Comment (nicoddemus) [inline setup.py:4]: Why is `name` necessary here? It is already present on `pyproject.toml`, I think we don't need to repeat it here?

```suggestion
setup()
```

**Review by nicoddemus — APPROVED** [review]

File: setup.py (hunk 3 above)
Comment (marcelotrevisani) [reply setup.py:4]: It is not necessary, it can be removed, I just let it to show the project name :)

**Pull request merged**